File: python/calculator.py
class Calculator:
    def __init__(self, authorizer):
        self.authorizer = authorizer

    def divide(self, numerator, denominator):
        authorized = self.authorizer.authorize()
        # Bug! Should be `if not authorized`
        if authorized:
            raise Exception("Not authorized")
        return numerator / denominator

File: js/calculator.js
class Calculator {
  constructor (authorizer) {
    this.authorizer = authorizer
  }

  divide (numerator, denominator) {
    const authorized = this.authorizer.authorize()
    // Bug! Sholud be `if(!authorized)`
    if (authorized) {
      throw new Error('Not authorized')
    }
    return numerator / denominator
  }
}

module.exports = Calculator
